nzedb/NameFixer.php

```diff
@@ -1494,4 +1578,38 @@ public function fileCheck($release, $echo, $type, $namestatus, $show)
 			}
 		}
 	}
+
+	/**
+	 * Look for a name based on mediainfo xml Unique_ID.
+	 *
+	 * @param         $release
+	 * @param         $release
+	 * @param boolean $echo
+	 * @param string  $type
+	 * @param         $namestatus
+	 * @param         $show
+	 */
+	public function uidCheck($release, $echo, $type, $namestatus, $show)
+	{
+		if ($this->done === false && $this->relid !== $release["releases_id"]) {
+			$result = $this->pdo->queryExec(
+				sprintf('
+						SELECT r.id AS releases_id, r.size AS relsize, r.name AS textstring, r.searchname AS searchname, r.predb_id, HEX(ru.uniqueid) AS uid
+						FROM releases r
+						INNER JOIN release_unique ru ON (ru.releases_id = r.id)
+						WHERE (r.isrenamed = %s OR r.categories_id NOT IN(%s)) AND r.predb_id > 0',
+						self::IS_RENAMED_DONE,
+						$this->othercats
+						)
+					);
+			foreach ($result as $rel) {
+				$percentage = $rel['relsize'] * (5/100);
+				$minSize = $rel['relsize'] - $percentage;
+				$maxSize = $rel['relsize'] + $percentage;
+				if ($rel['uid'] === $release['uid'] && ($minSize <= $release['relsize'] && $maxSize >= $release['relsize'])) {
```

Comment:
After removing the foreach above, this should be the only if conditional left after the query.  Simply say:

```php
if ($result !== false) {
   $this->updateRelease($release, $rel['searchname'], $method = "uidCheck: Unique_ID", $echo, $type, $namestatus, $show, $rel['predb_id']);
}
```